0.1/604 options screen menu

charlesstrange2 opened the pull request.

charlesstrange2 committed:
Create init Route
Create init Controller

charlesstrange2 committed:
typos

charlesstrange2 committed:
Route based auto auth, because no login pages at the moment.

charlesstrange2 committed:
remove debug statement

charlesstrange2 committed:
Create a basic template from options.html
Fix asset moans from IDE

charlesstrange2 committed:
Fix asset in masthead. it's asset() not assets()!

katjam commented on app/Http/Controllers/Service/DashboardController.php:
```diff
@@ -0,0 +1,22 @@
+        $user = Auth::user();
+        $data = [
+            "user_name" => $user->name,
+            "centre_name" => $user->centre->name,
```
centre_id is nullable. Do we need to make sure this user has one?

charlesstrange2 replied:
We do, I'll change it.

katjam commented on app/Http/Middleware/RouteAutoAuthenticate.php:
```diff
@@ -0,0 +1,32 @@
+    public function handle($request, Closure $next, $guard = null)
+    {
+        // check request for a user param; use #1 if there isn't one.
+        $user_id = ($request->query('user')) ? $request->query('user') : 1;
```
Who is user one? Do we want to hard code that in?

charlesstrange2 replied:
User one is setup by the supporting seeds. System will throw an exception if user one doesn't exist.
At this point i just want a default user with the option to fiddle the url to find another one.

katjam commented on app/User.php:
```diff
@@ -27,8 +27,14 @@ class User extends Authenticatable
         'password', 'remember_token',
     ];
 
+
```
Extra space.

charlesstrange2 replied:
will remove.

katjam commented on app/User.php:
```diff
@@ -27,8 +27,14 @@ class User extends Authenticatable
         return $this->hasMany('App\Note');
     }
+
+    public function centre()
```
Boring of me to say - but missing doc comments.

charlesstrange2 replied:
okedoke.

katjam reviewed (commented):
Pretty close.

charlesstrange2 committed:
Add docs
Fix typos and spacing
Add method for accepting null user->centre relationships.

katjam reviewed (approved):
Looks good to me! :lollipop:

charlesstrange2 committed:
Merge branch '0.1/release' into 0.1/604-options-screen-menu
